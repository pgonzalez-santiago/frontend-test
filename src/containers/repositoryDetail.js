import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'

// Redux
import RepositoryDetailActions from '../store/reducers/repositoryDetail'
// Conatiners
import ContributorsList from './contributorsList'
// Components
import Spinner from '../components/spinner'

// External dependencies
import { withGetScreen } from 'react-getscreen'
import styled from 'styled-components'
import Colors from '../assets/colors'
// Languages
import I18n from '../assets/lang'

const Wrapper = styled.div`
  position: relative;
  top: 35px;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 30px;
  padding-top: 0;
  color: ${Colors.backgroundColor};
`

const Container = styled.div`
  width: ${(props) => props.isMobile ? 'auto' : '70%'};
  overflow: auto;
  position: absolute;
  top: ${(props) => props.isMobile ? '80px' : 0};
  right: 0;
  left: 0;
  bottom: 0;
`

const Title = styled.h1`
  padding-bottom: 20px;
  border-bottom: 2px solid;
  text-transform: uppercase;
  padding-right: 35px;
`

const Detail = styled.div`
  padding-top: 10px;
  padding-bottom: 20px;
`

const MessageWrapper = styled.div`
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  color: ${Colors.primaryColor};
  font-size: ${(props) => props.isMobile ? '20px' : '30px'};;
  text-transform: uppercase;
  padding: 20px;
`

class RepositoryDetail extends PureComponent {
  componentDidMount () {
    const { getDetail, repoName } = this.props

    if (repoName) {
      // Get repository detailed data when component is mounted
      getDetail(`owner: "facebook", name: "${repoName}"`)
    }
  }

  componentDidUpdate (prevProps) {
    const { getDetail, repoName } = this.props

    // If repository changes get the new repository detailed data
    if (repoName !== prevProps.repoName) {
      getDetail(`owner: "facebook", name: "${repoName}"`)
    }
  }

  renderContent () {
    const { detail, error, fetching, isMobile, repoName } = this.props

    // If something goes wrong in the request render a message
    if (error) {
      return (
        <MessageWrapper isMobile={isMobile()}>
          <span>{I18n.t('errorMessage')}</span>
        </MessageWrapper>
      )
    }

    // Render a spinner while repository detail data is being fetched
    if (fetching) {
      return (
        <MessageWrapper isMobile={isMobile()}>
          <Spinner
            color={Colors.primaryColor}
            size={50}/>
        </MessageWrapper>
      )
    }

    // If a repo is selected render the data
    if (detail) {
      return (
        <Wrapper>
          <Title>
            {I18n.t('detailTitle')}
          </Title>
          <Detail>
            <p>{I18n.t('Name')}{detail.name}</p>
            <p>{I18n.t('Description')}{detail.description}</p>
            <p>{I18n.t('Forks')}{detail.forks}</p>
            <p>{I18n.t('Issues')} {detail.issues}</p>
            <p>{I18n.t('Pull requests')}{detail.pullRequests}</p>
            <p>{I18n.t('Release')}{detail.releases}</p>
            <p>{I18n.t('Stars')}{detail.stars}</p>
            <p>{I18n.t('Watchers')}{detail.watchers}</p>
          </Detail>
          <ContributorsList repoName={repoName}/>
        </Wrapper>
      )
    }

    // IF there is no repository selected display a message to select a repository
    return (
      <MessageWrapper isMobile={isMobile()}>
        <span>{I18n.t('selectRepo')}</span>
      </MessageWrapper>
    )
  }

  render () {
    const { isMobile } = this.props

    return (
      <Container id="page-wrap" isMobile={isMobile()}>
        {this.renderContent()}
      </Container>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    detail:   state.repositoryDetail.repository,
    error:    state.repositoryDetail.error,
    fetching: state.repositoryDetail.fetching,
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    getDetail: (query) => dispatch(RepositoryDetailActions.request(query)),
  }
}

const RespositoryDetailCContainer = withGetScreen(RepositoryDetail)

export default connect(mapStateToProps, mapDispatchToProps)(RespositoryDetailCContainer)

RepositoryDetail.propTypes = {
  detail: PropTypes.shape({
    description:  PropTypes.string.isRequired,
    forks:        PropTypes.number.isRequired,
    issues:       PropTypes.number.isRequired,
    name:         PropTypes.string.isRequired,
    pullRequests: PropTypes.number.isRequired,
    releases:     PropTypes.number.isRequired,
    stars:        PropTypes.number.isRequired,
    watchers:     PropTypes.number.isRequired,
  }),
  error:     PropTypes.bool.isRequired,
  fetching:  PropTypes.bool.isRequired,
  getDetail: PropTypes.func.isRequired,
  isMobile:  PropTypes.func.isRequired,
  repoName:  PropTypes.string,
}
