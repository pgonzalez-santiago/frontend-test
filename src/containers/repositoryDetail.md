# Repository Detail

----
## Props

* **repoName**             String: selected repo.

----
## Usage

```html
<RepositoryDetail repoName={'react'} />
```
