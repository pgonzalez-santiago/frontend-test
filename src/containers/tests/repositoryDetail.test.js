import React from 'react'
import { RepositoryDetail } from '../repositoryDetail'
import { shallow } from 'enzyme'

describe('RepositoryDetail', () => {
  it('RepositoryDetail Dom structure has not change when error', () => {
    const wrapper = shallow(
      <RepositoryDetail
        detail={null}
        error={true}
        fetching={false}
        getDetail={() => null}
        isMobile={() => false}
        repoName={'react'}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('RepositoryDetail Dom structure has not change when fetching', () => {
    const wrapper = shallow(
      <RepositoryDetail
        detail={null}
        error={false}
        fetching={true}
        getDetail={() => null}
        isMobile={() => false}
        repoName={'react'}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('RepositoryDetail Dom structure has not change when render repository detail', () => {
    const wrapper = shallow(
      <RepositoryDetail
        detail={{
          description: 'A framework for building native apps with React.',
          homepageUrl: 'https://facebook.github.io/react-native/',
          name: 'react-native',
          projectsUrl: 'https://github.com/facebook/react-native/projects',
          forks: 15461,
          issues: 14658,
          pullRequests: 7625,
          releases: 15461,
          stars: 71397,
          watchers: 3712
        }}
        error={false}
        fetching={false}
        getDetail={() => null}
        isMobile={() => false}
        repoName={'react'}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
