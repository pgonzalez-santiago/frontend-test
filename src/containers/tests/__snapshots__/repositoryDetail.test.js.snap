// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RepositoryDetail RepositoryDetail Dom structure has not change when error 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RepositoryDetail
    detail={null}
    error={true}
    fetching={false}
    getDetail={[Function]}
    isMobile={[Function]}
    repoName="react"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef
        isMobile={false}
      >
        <span>
          Ups something went wrong!!
        </span>
      </ForwardRef>,
      "id": "page-wrap",
      "isMobile": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <span>
          Ups something went wrong!!
        </span>,
        "isMobile": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Ups something went wrong!!",
        },
        "ref": null,
        "rendered": "Ups something went wrong!!",
        "type": "span",
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-iwsKbI",
          "isStatic": false,
          "rules": Array [
            "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  color: ",
            "#b8b7ad",
            ";
  font-size: ",
            [Function],
            ";;
  text-transform: uppercase;
  padding: 20px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-iwsKbI",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gzVnrw",
        "isStatic": false,
        "rules": Array [
          "
  width: ",
          [Function],
          ";
  overflow: auto;
  position: absolute;
  top: ",
          [Function],
          ";
  right: 0;
  left: 0;
  bottom: 0;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-gzVnrw",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          isMobile={false}
        >
          <span>
            Ups something went wrong!!
          </span>
        </ForwardRef>,
        "id": "page-wrap",
        "isMobile": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <span>
            Ups something went wrong!!
          </span>,
          "isMobile": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Ups something went wrong!!",
          },
          "ref": null,
          "rendered": "Ups something went wrong!!",
          "type": "span",
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-iwsKbI",
            "isStatic": false,
            "rules": Array [
              "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  color: ",
              "#b8b7ad",
              ";
  font-size: ",
              [Function],
              ";;
  text-transform: uppercase;
  padding: 20px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-iwsKbI",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gzVnrw",
          "isStatic": false,
          "rules": Array [
            "
  width: ",
            [Function],
            ";
  overflow: auto;
  position: absolute;
  top: ",
            [Function],
            ";
  right: 0;
  left: 0;
  bottom: 0;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-gzVnrw",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`RepositoryDetail RepositoryDetail Dom structure has not change when fetching 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RepositoryDetail
    detail={null}
    error={false}
    fetching={true}
    getDetail={[Function]}
    isMobile={[Function]}
    repoName="react"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef
        isMobile={false}
      >
        <Spinner
          color="#b8b7ad"
          size={50}
        />
      </ForwardRef>,
      "id": "page-wrap",
      "isMobile": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Spinner
          color="#b8b7ad"
          size={50}
        />,
        "isMobile": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "#b8b7ad",
          "size": 50,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-iwsKbI",
          "isStatic": false,
          "rules": Array [
            "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  color: ",
            "#b8b7ad",
            ";
  font-size: ",
            [Function],
            ";;
  text-transform: uppercase;
  padding: 20px;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-iwsKbI",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gzVnrw",
        "isStatic": false,
        "rules": Array [
          "
  width: ",
          [Function],
          ";
  overflow: auto;
  position: absolute;
  top: ",
          [Function],
          ";
  right: 0;
  left: 0;
  bottom: 0;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-gzVnrw",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef
          isMobile={false}
        >
          <Spinner
            color="#b8b7ad"
            size={50}
          />
        </ForwardRef>,
        "id": "page-wrap",
        "isMobile": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": <Spinner
            color="#b8b7ad"
            size={50}
          />,
          "isMobile": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "#b8b7ad",
            "size": 50,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-iwsKbI",
            "isStatic": false,
            "rules": Array [
              "
  display: flex;
  height: 100%;
  align-items: center;
  justify-content: center;
  color: ",
              "#b8b7ad",
              ";
  font-size: ",
              [Function],
              ";;
  text-transform: uppercase;
  padding: 20px;
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-iwsKbI",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gzVnrw",
          "isStatic": false,
          "rules": Array [
            "
  width: ",
            [Function],
            ";
  overflow: auto;
  position: absolute;
  top: ",
            [Function],
            ";
  right: 0;
  left: 0;
  bottom: 0;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-gzVnrw",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`RepositoryDetail RepositoryDetail Dom structure has not change when render repository detail 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RepositoryDetail
    detail={
      Object {
        "description": "A framework for building native apps with React.",
        "forks": 15461,
        "homepageUrl": "https://facebook.github.io/react-native/",
        "issues": 14658,
        "name": "react-native",
        "projectsUrl": "https://github.com/facebook/react-native/projects",
        "pullRequests": 7625,
        "releases": 15461,
        "stars": 71397,
        "watchers": 3712,
      }
    }
    error={false}
    fetching={false}
    getDetail={[Function]}
    isMobile={[Function]}
    repoName="react"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <ForwardRef>
        <ForwardRef>
          Repository details
        </ForwardRef>
        <ForwardRef>
          <p>
            Name: 
            react-native
          </p>
          <p>
            Description: 
            A framework for building native apps with React.
          </p>
          <p>
            Forks: 
            15461
          </p>
          <p>
            Issues: 
             
            14658
          </p>
          <p>
            Pull requests: 
            7625
          </p>
          <p>
            Release: 
            15461
          </p>
          <p>
            Stars: 
            71397
          </p>
          <p>
            Watchers: 
            3712
          </p>
        </ForwardRef>
        <Connect(ContributorsList)
          repoName="react"
        />
      </ForwardRef>,
      "id": "page-wrap",
      "isMobile": false,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Repository details
          </ForwardRef>,
          <ForwardRef>
            <p>
              Name: 
              react-native
            </p>
            <p>
              Description: 
              A framework for building native apps with React.
            </p>
            <p>
              Forks: 
              15461
            </p>
            <p>
              Issues: 
               
              14658
            </p>
            <p>
              Pull requests: 
              7625
            </p>
            <p>
              Release: 
              15461
            </p>
            <p>
              Stars: 
              71397
            </p>
            <p>
              Watchers: 
              3712
            </p>
          </ForwardRef>,
          <Connect(ContributorsList)
            repoName="react"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Repository details",
          },
          "ref": null,
          "rendered": "Repository details",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-htoDjs",
              "isStatic": true,
              "rules": Array [
                "
  padding-bottom: 20px;
  border-bottom: 2px solid;
  text-transform: uppercase;
  padding-right: 35px;
",
              ],
            },
            "displayName": "styled.h1",
            "render": [Function],
            "styledComponentId": "sc-htoDjs",
            "target": "h1",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <p>
                Name: 
                react-native
              </p>,
              <p>
                Description: 
                A framework for building native apps with React.
              </p>,
              <p>
                Forks: 
                15461
              </p>,
              <p>
                Issues: 
                 
                14658
              </p>,
              <p>
                Pull requests: 
                7625
              </p>,
              <p>
                Release: 
                15461
              </p>,
              <p>
                Stars: 
                71397
              </p>,
              <p>
                Watchers: 
                3712
              </p>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Name: ",
                  "react-native",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Name: ",
                "react-native",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Description: ",
                  "A framework for building native apps with React.",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Description: ",
                "A framework for building native apps with React.",
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Forks: ",
                  15461,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Forks: ",
                15461,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Issues: ",
                  " ",
                  14658,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Issues: ",
                " ",
                14658,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Pull requests: ",
                  7625,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Pull requests: ",
                7625,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Release: ",
                  15461,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Release: ",
                15461,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Stars: ",
                  71397,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Stars: ",
                71397,
              ],
              "type": "p",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Watchers: ",
                  3712,
                ],
              },
              "ref": null,
              "rendered": Array [
                "Watchers: ",
                3712,
              ],
              "type": "p",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-dnqmqq",
              "isStatic": true,
              "rules": Array [
                "
  padding-top: 10px;
  padding-bottom: 20px;
",
              ],
            },
            "displayName": "styled.div",
            "render": [Function],
            "styledComponentId": "sc-dnqmqq",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "repoName": "react",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bZQynM",
          "isStatic": true,
          "rules": Array [
            "
  position: relative;
  top: 35px;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 30px;
  padding-top: 0;
  color: ",
            "#24292e",
            ";
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bZQynM",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gzVnrw",
        "isStatic": false,
        "rules": Array [
          "
  width: ",
          [Function],
          ";
  overflow: auto;
  position: absolute;
  top: ",
          [Function],
          ";
  right: 0;
  left: 0;
  bottom: 0;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-gzVnrw",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <ForwardRef>
          <ForwardRef>
            Repository details
          </ForwardRef>
          <ForwardRef>
            <p>
              Name: 
              react-native
            </p>
            <p>
              Description: 
              A framework for building native apps with React.
            </p>
            <p>
              Forks: 
              15461
            </p>
            <p>
              Issues: 
               
              14658
            </p>
            <p>
              Pull requests: 
              7625
            </p>
            <p>
              Release: 
              15461
            </p>
            <p>
              Stars: 
              71397
            </p>
            <p>
              Watchers: 
              3712
            </p>
          </ForwardRef>
          <Connect(ContributorsList)
            repoName="react"
          />
        </ForwardRef>,
        "id": "page-wrap",
        "isMobile": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef>
              Repository details
            </ForwardRef>,
            <ForwardRef>
              <p>
                Name: 
                react-native
              </p>
              <p>
                Description: 
                A framework for building native apps with React.
              </p>
              <p>
                Forks: 
                15461
              </p>
              <p>
                Issues: 
                 
                14658
              </p>
              <p>
                Pull requests: 
                7625
              </p>
              <p>
                Release: 
                15461
              </p>
              <p>
                Stars: 
                71397
              </p>
              <p>
                Watchers: 
                3712
              </p>
            </ForwardRef>,
            <Connect(ContributorsList)
              repoName="react"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": "Repository details",
            },
            "ref": null,
            "rendered": "Repository details",
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-htoDjs",
                "isStatic": true,
                "rules": Array [
                  "
  padding-bottom: 20px;
  border-bottom: 2px solid;
  text-transform: uppercase;
  padding-right: 35px;
",
                ],
              },
              "displayName": "styled.h1",
              "render": [Function],
              "styledComponentId": "sc-htoDjs",
              "target": "h1",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": Array [
                <p>
                  Name: 
                  react-native
                </p>,
                <p>
                  Description: 
                  A framework for building native apps with React.
                </p>,
                <p>
                  Forks: 
                  15461
                </p>,
                <p>
                  Issues: 
                   
                  14658
                </p>,
                <p>
                  Pull requests: 
                  7625
                </p>,
                <p>
                  Release: 
                  15461
                </p>,
                <p>
                  Stars: 
                  71397
                </p>,
                <p>
                  Watchers: 
                  3712
                </p>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Name: ",
                    "react-native",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Name: ",
                  "react-native",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Description: ",
                    "A framework for building native apps with React.",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Description: ",
                  "A framework for building native apps with React.",
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Forks: ",
                    15461,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Forks: ",
                  15461,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Issues: ",
                    " ",
                    14658,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Issues: ",
                  " ",
                  14658,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Pull requests: ",
                    7625,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Pull requests: ",
                  7625,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Release: ",
                    15461,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Release: ",
                  15461,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Stars: ",
                    71397,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Stars: ",
                  71397,
                ],
                "type": "p",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Watchers: ",
                    3712,
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Watchers: ",
                  3712,
                ],
                "type": "p",
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-dnqmqq",
                "isStatic": true,
                "rules": Array [
                  "
  padding-top: 10px;
  padding-bottom: 20px;
",
                ],
              },
              "displayName": "styled.div",
              "render": [Function],
              "styledComponentId": "sc-dnqmqq",
              "target": "div",
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "repoName": "react",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bZQynM",
            "isStatic": true,
            "rules": Array [
              "
  position: relative;
  top: 35px;
  bottom: 0;
  left: 0;
  right: 0;
  padding: 30px;
  padding-top: 0;
  color: ",
              "#24292e",
              ";
",
            ],
          },
          "displayName": "styled.div",
          "render": [Function],
          "styledComponentId": "sc-bZQynM",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gzVnrw",
          "isStatic": false,
          "rules": Array [
            "
  width: ",
            [Function],
            ";
  overflow: auto;
  position: absolute;
  top: ",
            [Function],
            ";
  right: 0;
  left: 0;
  bottom: 0;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-gzVnrw",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
